R/tm_g_events_term_id.R

```diff
@@ -287,34 +280,39 @@ srv_g_events_term_id <- function(id,
           selected = choices[trt_index],
           choices = choices
         )
-      }
+      },
+      ignoreNULL = TRUE
     )
 
     output_q <- reactive({
       ANL <- data[[dataname]]() # nolint
 
-      validate(need(iv$is_valid(), "Misspecification error: please observe red flags in the encodings."))
-
-      validate(need(
-        is.factor(ANL[[input$arm_var]]),
-        "Selected arm variable needs to be a factor. Contact an app developer."
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("term", shinyvalidate::sv_required(
+        message = "Term Variable is required"
       ))
-
-      iv_comp <- shinyvalidate::InputValidator$new()
-      iv_comp$add_rule("arm_trt", shinyvalidate::sv_not_equal(
+      iv$add_rule("arm_var", shinyvalidate::sv_required(
+        message = "Arm Variable is required"
+      ))
+      iv$add_rule("arm_var", ~ if (!is.factor(ANL[[req(.)]])) {
+        "Arm Var must be a factor variable, contact developer"
+      })
+      iv$add_rule("arm_trt", shinyvalidate::sv_not_equal(
         input$arm_ref,
-        message_fmt = "Must not be equal to Control"
+        message_fmt = "Control and Treatment must be different"
       ))
-      iv_comp$add_rule("arm_ref", shinyvalidate::sv_not_equal(
+      iv$add_rule("arm_ref", shinyvalidate::sv_not_equal(
         input$arm_trt,
-        message_fmt = "Must not be equal to Treatment"
+        message_fmt = "Control and Treatment must be different"
       ))
-      iv_comp$enable()
-      validate(need(iv_comp$is_valid(), "Misspecification error: please observe red flags in the encodings."))
+      iv$enable()
+
+      teal::validate_inputs(iv)
 
       validate(need(
-        all(c(input$arm_trt, input$arm_ref) %in% unique(ANL[[input$arm_var]])),
-        "Cannot generate plot. The dataset does not contain subjects from both the control and treatment arms."
+        input$arm_trt %in% unique(ANL[[req(input$arm_var)]]) &&
+          input$arm_ref %in% unique(ANL[[req(input$arm_var)]]),
+        "Cannot generate plot. No subjects in both Control and Treatment arms."
```

Comment:
I'm not sure "No subjects in both Control and Treatment arms." is quite clear I would leave the message as it was

Reply:
restoerd